api/pkg/db/migration/202107221500_add_ondelete_constraints.go

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+// Drops existing foreign key constraints and create the same constraints
+// in order to add another OnDelete constraints to foriegn key
+func addOnDeleteConstraints(log *log.Logger) *gormigrate.Migration {
+
+	return &gormigrate.Migration{
+		ID: "202107221500_add_ondelete_constraints",
+		Migrate: func(db *gorm.DB) error {
+			txn := db.Begin()
+
+			err := addOnDelete(txn, log)
+			if err != nil {
+				txn.Rollback()
+				log.Error(err)
+				return err
+			}
+			txn.Commit()
+			return nil
+		},
+	}
+}
+
+func addOnDelete(txn *gorm.DB, log *log.Logger) error {
+	if err := txn.Migrator().DropConstraint(&model.ResourceCategory{}, "fk_resource_categories_resource"); err != nil {
+		log.Error(err)
+		return err
+	}
+
+	if err := txn.Exec("ALTER TABLE resource_categories ADD CONSTRAINT fk_resource_categories_resource  FOREIGN KEY (resource_id) REFERENCES resources(id) ON DELETE CASCADE;").Error; err != nil {
```

Comment:
Not mandatory just a thought can't we use update query from gorm instead of writing the whole query, else the patch looks fine to me

Reply:
For example 
Drop Constraint 
update column from table by adding foreign key constraints with on Delete cascade 
For ref : https://gorm.io/docs/update.html#Update-single-column

Reply:
No, we can't do with gorm and with gorm CreateConstraint does't create